src/components/DeviceListTable.tsx

```diff
@@ -0,0 +1,100 @@
+import { FC } from "react";
+import { MainTable } from "@canonical/react-components";
+import { isRootDisk } from "util/instanceValidation";
+import { FormDevice } from "util/formDevices";
+import ResourceLink from "components/ResourceLink";
+import { isOtherDevice } from "util/devices";
+import DeviceDetails from "./DeviceDetails";
+import { useParams } from "react-router-dom";
+import { LxdDeviceValue, LxdDevices } from "types/device";
+
+interface Props {
+  configBaseURL: string;
+  devices: LxdDevices;
+}
+
+const DeviceListTable: FC<Props> = ({ configBaseURL, devices }) => {
+  const { project } = useParams<{ project: string }>();
+
+  const compareDevices = (
+    a: [string, LxdDeviceValue],
+    b: [string, LxdDeviceValue],
+  ) => {
+    const nameA = a[0].toLowerCase();
+    const nameB = b[0].toLowerCase();
+    const typeA = a[1].type;
+    const typeB = b[1].type;
+    if (typeA === typeB) {
+      return nameA.localeCompare(nameB);
+    }
+
+    return typeA.localeCompare(typeB);
+  };
+
+  // Identify non-NIC devices
+  const overviewDevices = Object.entries(devices ?? {})
+    .filter(([_key, device]) => {
+      return device.type !== "nic" && device.type !== "none";
+    })
+    // Sort devices by type and name
+    .sort(compareDevices);
```

Comment:
```suggestion
    .sort(byTypeAndName);
```
By naming the function, you can make the code speaking and avoid the need for a comment